truffleruby/src/main/c/openssl/ossl_asn1.c

```diff
@@ -845,7 +845,8 @@ int_ossl_asn1_decode0_prim(unsigned char **pp, long length, long hlen, int tag,
     *pp += hlen + length;
     *num_read = hlen + length;
 
-    if (tc == sUNIVERSAL && tag < ossl_asn1_info_size && ossl_asn1_info_klass[tag]) { // ossl_asn1_info[tag].klass) {
+    // Using truffle_is_truffle_object to workaround issue casting truffle object to int for truth comparison
```

Comment:
> So it's testing the truthiness of a TruffleObject which is problematic?

Yes.

> Why does testing the truthiness requires the address (what the error message says)?
I'm speculating that it's trying to copy to native since integer comparison isn't supported with a TruffleObject. 

From what I understand it's trying to copy a TruffleObject `OpenSSL::ASN1::Boolean` into the native heap as an int (since integer comparison isn't a supported operation). But I could be misunderstanding this.

@chrisseaton Does this sound right? Could you clarify?

References:
https://github.com/graalvm/truffleruby/blob/master/doc/user/compatibility.md#c-extension-compatibility
https://github.com/graalvm/truffleruby/blob/master/doc/contributor/cexts.md#array-and-string-pointers